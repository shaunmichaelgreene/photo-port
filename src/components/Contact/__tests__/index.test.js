import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Contact from '..';

afterEach(cleanup);

describe('Contact component', () => {
    it('renders', () => {
        render(<Contact />)
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Contact />);
        expect(asFragment()).toMatchSnapshot();
    });
})

describe('Contact Form says "Contact Me"', () => {
    it('inserts text content into the contact form', () => {
        const { getByTestId } = render(<Contact />);
        expect(getByTestId('contact-form')).toHaveTextContent('Contact Me');
    });
})

describe('button text content is "Submit"', () => {
    it('inserts text content into the contact form submit button', () => {
        const { getByTestId } = render(<Contact />);
        expect(getByTestId('submit-button')).toHaveTextContent('Submit');
    });
})
