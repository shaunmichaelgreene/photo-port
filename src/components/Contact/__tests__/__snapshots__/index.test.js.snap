// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1
      data-testid="contact-form"
    >
      Contact Me
    </h1>
    <form
      id="contact-form"
    >
      <div>
        <label
          for="name"
        >
          Name:
        </label>
        <input
          name="name"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email Address:
        </label>
        <input
          name="email"
          type="email"
          value=""
        />
      </div>
      <div>
        <label
          for="message"
        >
          Message:
        </label>
        <textarea
          name="message"
          rows="5"
        />
      </div>
      <button
        data-testid="submit-button"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
